File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripService.java
package org.craftedsw.tripservicekata.trip;

import java.util.ArrayList;
import java.util.List;

import org.craftedsw.tripservicekata.exception.UserNotLoggedInException;
import org.craftedsw.tripservicekata.user.User;
import org.craftedsw.tripservicekata.user.UserSession;

public class TripService {

	public List<Trip> getTripsByUser(User user) throws UserNotLoggedInException {
		User loggedUser = getLoggedUser();
		assertNotGuest(loggedUser);
		if (user.isFriendsWith(loggedUser)) {
			return findTripsByUser(user);
		} else {
			return noTrips();
		}
	}

	private void assertNotGuest(User loggedUser) {
		if (loggedUser == null) {
			throw new UserNotLoggedInException();
		}
	}

	private ArrayList<Trip> noTrips() {
		return new ArrayList<Trip>();
	}

	protected List<Trip> findTripsByUser(User user) {
		return TripDAO.findTripsByUser(user);
	}

	protected User getLoggedUser() {
		return UserSession.getInstance().getLoggedUser();
	}

}

File: javascript/src/TripService.js
"use strict";

let UserSession = require('./UserSession');
let TripDAO = require('./TripDAO');

class TripService {

    getTripsByUser(user) {
        let loggedUser = this.getLoggedUser();
        this.assertNotGuest(loggedUser);

        let noTrips = [];
        return user.isFriendsWith(loggedUser) ? this.findTripsBy(user) : noTrips;
    }

    assertNotGuest(loggedUser) {
        if (loggedUser == null) {
            throw new Error('User not logged in.');
        }
    }

    findTripsBy(user) {
        return TripDAO.findTripsByUser(user);
    }

    getLoggedUser() {
        let loggedUser = UserSession.getLoggedUser();
        return loggedUser;
    }
}

module.exports = TripService
